### ISSUE TITLE
Implement Guardian Codebase

### ISSUE THREAD

**Issue by TinyPandas** [RSA-Internal/PandaProtector#7]
TODO
- [x] Split this into separate issues before starting.
- [ ] Implement mute
- [x] Implement warn
- [ ] Implement ban
- [ ] Implement tempban

**Comment by M39a9am3R** [comment]
Will start work on commands when Discord.JS v13 releases. Intended commands and structure.

- [ ] `/ban perm [user] [reason] [messageId]` - Will put user into a temporary ban state of 31-days until confirmed by a Super Staff within that time frame where the permanent ban will take effect (2-staff system).
  - `[user]` ID, mention, or name of user. *required*.
  - `[reason]` Reason for the action. *required*.
  - `[messageId]` Context or offending message resulting in action. Default "". *optional*.
- [ ] `/ban temp [user] [reason] [length] [messageId]` - Will remove the user's access to all channels in the server except for an appeals channel notifying the user they can message the bot to request an appeal.
  - `[user]` *required* , `[reason]` *required* , `[messageId]` *optional*.
  - `[length]` In context of days. 14 default, 1 minimum, 31 maximum. *optional*.
- [ ] `/mute [user] [reason] [messageId] [length]` - Will prevent the user from chatting in all channels in the server, however, will still grant the user to view the contents of the channels.
  - `[user]` *required*, `[reason]` *required*, `[messageId]` *optional*.
  - `[length]` In context of minutes. 30 * 2 ** muteCountInThirtyDays default, 5 minimum, 1440 maximum. *optional*.
    - This would make it where a person receiving their first mute in a month will get a 30 minute mute. Second will get a 60 minute mute. Third will get a 120 minute. Etc.
- [ ] `/warn [user] [reason] [messageId]` - Will notify the user they have received a warning from the server.
  - `[user]` *required*, `[reason]` *required*, `[messageId]` *optional*.
- [ ] `/note [user] [reason]` - Will add a note for moderators to see. Will not notify the user which the note was placed.
  - `[user]` *required*, `[reason]` *required*.
- [ ] `/history [user]` - Will provide an HTML document of all infractions and notes of the user, should there be more than 3 records. If there are fewer than three records, will be posted as an embed.
  - `[user]` *required*.
- [ ] `/purge [timeframe] [user]` - Will remove messages in a channel from the prior amount of time specified in the timeframe, user argument to target a specific user's messages.
  - `[user]` *optional*.
  - `[timeframe]` In context of minutes, minimum 1 minute, maximum 1440 minutes. *required*.

When a messageId is specified the message will be stored to the database and removed from the server.
